pkg/asset/installconfig/gcp/gcp.go

```diff
@@ -82,13 +81,33 @@ func selectProject(ctx context.Context) (string, error) {
 	return selectedProject, err
 }
 
-func selectRegion(project string) (string, error) {
-	longRegions := make([]string, 0, len(validation.Regions))
-	shortRegions := make([]string, 0, len(validation.Regions))
-	for id, location := range validation.Regions {
-		longRegions = append(longRegions, fmt.Sprintf("%s (%s)", id, location))
-		shortRegions = append(shortRegions, id)
+func selectRegion(ctx context.Context, project string) (string, error) {
+
+	ssn, err := GetSession(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get session")
+	}
+
+	client := &Client{
+		ssn: ssn,
+	}
+
+	validRegions, err := client.GetProjectRegions(ctx, project)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get regions")
 	}
+
+	longRegions := make([]string, 0, len(validRegions))
+	shortRegions := make([]string, 0, len(validRegions))
+	for key, value := range validRegions {
+		shortRegions = append(shortRegions, key)
+		if value == "" {
+			longRegions = append(longRegions, key)
+		} else {
+			longRegions = append(longRegions, fmt.Sprintf("%s (%s)", key, value))
+		}
+	}
+
```

Comment:
nit: we can avoid the found logic with this? would reduce a for loop but maybe it's not worth reducing?
```suggestion
        defaultRegionName := ""
        defaultRegion := "us-central1"
	longRegions := make([]string, 0, len(validRegions))
	shortRegions := make([]string, 0, len(validRegions))
	for key, value := range validRegions {
		shortRegions = append(shortRegions, key)
		regionDesc := key
		if value != "" {
		        regionDesc := fmt.Sprintf("%s (%s)", key, value)
		}
		longRegions = append(longRegions, regionDesc)
		if key == defaultRegion {
		       defaultRegionName = regionDesc
		}
	}

```

Reply:
@rna-afk The reason that I did not choose the above was specifically because my solution will offer a region to the user even when the region does not have an exact known location (specific city/country). This will allow us to still offer the region as acceptable, and we will be able to find missing regions easier in the future.

Reply:
That sounds right but could you elaborate? I suggested this code hoping to exactly replicate your code logic but reduce the for loop count but I may have made some mistakes. IIUC your concern of adding a region to the list even if it does not have a proper description(longName) is taken care of by this code?